### PR TITLE
Add support for python3.10

### PR DESCRIPTION
**Pull request by amalshaji**

### CODE REVIEW

**Comment by amalshaji** [comment]
@carltongibson Can you approve my workflow? I want to see if python3.10 works!!

**Review by carltongibson — COMMENTED** [review]
You'll also need to adjust the GHA workflow: 

https://github.com/carltongibson/django-filter/blob/3635b2b67c110627e74404603330583df0000a44/.github/workflows/tests.yml#L11

See 2ebce743d5facbe4fe628b79fa916a3f5ef09c21 for additional extras, like trove classifiers, and updating the README, and so on.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9]
+        python-version: [3.6, 3.7, 3.8, 3.9, "3.10"]
```
Comment (carltongibson) [inline .github/workflows/tests.yml:11]: We may as well make all these strings whilst we're here.

**Review by carltongibson — COMMENTED** [review]
This is looking good @amalshaji

**Comment by carltongibson** [comment]
... except the CI failed. (Looks like a config error. 🤔  )

**Comment by amalshaji** [comment]
https://github.com/amalshaji/django-filter/blob/018ff908eabeddbe6efa60a292d73ad022ee5c4c/.github/workflows/tests.yml#L29

```shell
✗ echo py$(python --version 2>&1 | cut -c 8,10)
py31
```

Tox expects it to be [py310](https://tox.wiki/en/latest/changelog.html#id150)

**Comment by amalshaji** [comment]
How about this?

```shell
py$(python --version 2>&1 | cut -c 8-11 | awk -F '.' '{print $1$2}')
```

Works for both single and double-digit minor versions

```shell
✗ echo py$(python --version | cut -c 8-11 | awk -F '.' '{print $1$2}')
py310
✗ echo py$(echo "Python 3.9" | cut -c 8-11 | awk -F '.' '{print $1$2}')
py39
✗ echo py$(echo "Python 3.9.0" | cut -c 8-11 | awk -F '.' '{print $1$2}')
py39
✗ echo py$(echo "Python 3.6.5" | cut -c 8-11 | awk -F '.' '{print $1$2}')
py36
```

**Review by carltongibson — APPROVED** [review]
OK, thanks for this @amalshaji! 

Let's move to `tox-py` (which will itself be obsolete but is more up to date than `tox-factor`) 

👍

**Comment by amalshaji** [comment]
awesome 🚀

**Pull request merged**